mesonbuild/backend/ninjabackend.py

```diff
@@ -2134,8 +2134,8 @@ def _link_library(libname: str, static: bool, bundle: bool = False):
                 args += ['-C', 'link-arg=' + rpath_arg + ':' + rustc.get_target_libdir()]
 
         proc_macro_dylib_path = None
-        if getattr(target, 'rust_crate_type', '') == 'proc-macro':
-            proc_macro_dylib_path = os.path.abspath(os.path.join(target.subdir, target.get_filename()))
+        if cratetype == 'proc-macro':
+            proc_macro_dylib_path = self.get_target_filename_abs(target)
```

Comment:
... I wonder how this ever worked? os.path.abspath of the target subdir is not guaranteed to give a sensible answer, depending on what directory you run meson from: the source directory or the build directory or possibly neither one.

Reply:
Ok, I'll move it to a separte patch then